### PR TITLE
[UI] Make catalog toggling as a user preference

### PR DESCRIPTION
**Pull request by Chadha93**
Signed-off-by: Gaurav Chadha <gauravchadha1676@gmail.com>

**Description**
There are 3 toggles for toggling catalog content on or off:

1. The EYE icon on patterns / filters page
2. One on extensions page
3. One in user preferences

These 3 toggles' states are synced. But when you turn either of them off and refresh the page, it goes back to 'on' state.

This PR fixes #
Send all these toggles to user preference to maintained their states on refresh. [work in progress]

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Is this WIP/draft, @Chadha93 or ready for review?

**Comment by Chadha93** [comment]
> Is this WIP/draft, @Chadha93 or ready for review?

WIP

**Comment by leecalcote** [comment]
@gr455 do drive this forward.

**Comment by gr455** [comment]
@Chadha93 What all is left on the PR? I can try and help if you're stuck somewhere

**Comment by Chadha93** [comment]
> @Chadha93 What all is left on the PR? I can try and help if you're stuck somewhere

I'm bumping into some errors here. The extensions page and design updates are left. I'll push more updates tonight.

**Comment by Chadha93** [comment]
@theBeginner86 I'm stuck with the `e.then` in dataFetch on sending a request to `/api/user/prefs`.

**Comment by leecalcote** [comment]
@gr455 help, please.

**Comment by Chadha93** [comment]
@MUzairS15 Could you please help here? :)

File: ui/pages/extensions.js [hunk 1]
```diff
@@ -50,12 +51,59 @@ const styles = (theme) => ({
...
+                aria-label="Close"
+                color="inherit"
+                onClick={() => closeSnackbar(key)}
+              >
+                <CloseIcon />
+              </IconButton>
+            ),
+          });
+      },
+    )
```
Comment (theBeginner86) [inline ui/pages/extensions.js:106]:
```suggestion
     },
     err => console.error(err). // or any error handler function of your choice
    )
```

Comment (theBeginner86) [reply ui/pages/extensions.js:106]: The reason for that error is the required number of parameters required for the `data-fetch` function.
It requires 4 parameters:
`url`, `options = {}`, `successFn`, `errorFn`

Currently, `errorFn` was not being passed, resulting in an undefined error while running the `data-fetch` function.

Comment (theBeginner86) [reply ui/pages/extensions.js:106]: @Chadha93 will you please check after adding the err handler function?

Comment (Chadha93) [reply ui/pages/extensions.js:106]: Sure, @theBeginner86.

**Comment by Chadha93** [comment]
Adding video of the current state - 

https://user-images.githubusercontent.com/65453826/202864260-7d6fd172-e44d-4674-a82f-bbf4233ff68c.mp4

**Comment by leecalcote** [comment]
@gr455 what are we doing here?

**Comment by leecalcote** [comment]
@gr455 checking in, again...

**Review by gr455 — CHANGES_REQUESTED** [review]
@Chadha93 The prefs are being set, so that's good. But for me, everytime I go to the extensions page, the state of the extension is toggled (if it was on earlier it becomes off and vice versa) without my interacting with the toggle bar.

There are 3 perf requests being sent each time I visit the extensions page
1. True
2. True
3. (False if it was True and True if it was False)

**Comment by gr455** [comment]
> @Chadha93 The prefs are being set, so that's good. But for me, everytime I go to the extensions page, the state of the extension is toggled (if it was on earlier it becomes off and vice versa) without my interacting with the toggle bar.
> 
> There are 3 perf requests being sent each time I visit the extensions page
> 
> 1. True
> 2. True
> 3. (False if it was True and True if it was False)

Also since all of the times, the prefs are being set properly; I think the request is perfectly fine. Something seems wrong on the UI side

**Comment by gr455** [comment]
@Chadha93 Any updates on this? I think there is a little problem in the frontend logic. The user prefs seem to be working right.

**Comment by leecalcote** [comment]
@suhail34, will you give this a review?

**Comment by suhail34** [comment]
> @suhail34, will you give this a review?

Will need some time to understand the problem

**Comment by theBeginner86** [comment]
Demo:


https://user-images.githubusercontent.com/73700530/205587964-b8e519c1-a4da-45bf-9769-02a9e667389a.mov

**Review by Chadha93 — COMMENTED** [review]
LGTM, Thank you @theBeginner86 and @gr455

**Pull request merged**